src/components/main/sortBar/filters/copyLinkButton/CopyLinkButton.ts

```diff
@@ -0,0 +1,23 @@
+import './copyLinkButton.scss';
+
+class CopyLinkButton {
+  public render(): void {
+    const parentElement = document.querySelector('#copy-button') as HTMLElement;
+    const copyButton: HTMLButtonElement = document.createElement('button');
+    copyButton.classList.add('copy-button');
+    copyButton.innerText = 'Copy Link';
+
+    parentElement.innerHTML = '';
+    parentElement.append(copyButton);
+
+    copyButton.addEventListener('click', () => {
+      navigator.clipboard.writeText(window.location.href);
+      copyButton.innerText = 'Copied!';
+      setTimeout(() => {
+        copyButton.innerText = 'Copy Link';
+      }, 300);
```

Comment:
I saw `setTimeout` before with a different value. You should get rid of those magical numbers by moving those into a separate file